pandas/tests/indexing/test_indexing.py

```diff
@@ -5489,6 +5489,269 @@ def test_none_coercion_mixed_dtypes(self):
                                         strict_nan=True)
 
 
+class TestIndexingCallable(tm.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    def test_frame_loc_ix_callable(self):
+        # GH 11485
+        df = pd.DataFrame({'A': [1, 2, 3, 4], 'B': list('aabb'),
+                           'C': [1, 2, 3, 4]})
+        # iloc cannot use bool mask
```

Comment:
Just looked it up, and there is an extensive discussion of it in this issue: https://github.com/pydata/pandas/issues/3631 
and then a PR from @jreback (but a long time ago) that implemented the NotImplementedError as shown above: https://github.com/pydata/pandas/pull/3635/files

Summary is: it is not clear whether the boolean series should be realigned (since this is the rule when passing labeled indexers, but on the other hand `iloc` is position-based and not label-based), therefore it was chosen to raise an error in this case.